### ISSUE TITLE
Way to use module across processes

### ISSUE THREAD

**Issue by rwstauner** [kentnl/Test-File-ShareDir#2]
I'm not sure what a good interface would be, but at least some documentation could be useful.

In attempting to transition to this module I had one test script using `qx` to call a bin/ script
and I noticed the test share dir didn't work there.
So I had to dig into the module to understand how it works (by prepending a tempdir to `@INC`)
Then I constructed `Test::File::ShareDir::Object::Dist` manually, saved the tempdir path,
and for the tests that require it i prepend the tempdir path to `$ENV{PERL5LIB}` before using `qx`.

So, do you have a better suggestion than that?
If not, documenting the workaround (and more of the implementation) would be good, or maybe adding a method to do something similar if that seems useful.

Any thoughts?
Thanks!

**Comment by kentfredric** [comment]
Yeah, that's a tricky problem. Doing what you're doing with ::Object's does seem the right step for this.

The problem of course will be that Test::File::ShareDir won't always be implemented in terms of @INC hackery, that's a workaround for limitations in File::ShareDir, limitations I'm hoping to see dissolved at some stage. ( And that's kinda why I try not to draw too much attention to the technique )

I'd probably suggest something like

$ENV{PERL5OPT}="-MTest::File::ShareDir::Dist="    

With whatever magic you need bolted on there instead of passing the tempdir around. ( And that gets you the tempdir vivified in the child process ). ( NB: I have not myself verified doing it this way is possible, but I reason it should/could be )

As for documentation, man, that main module is getting long! I'm litterally running short of ideas on where to put new docs =)

Need to print a frigging book.

**Comment by rwstauner** [comment]
Thank you!  That is a much better idea.
I couldn't get the syntax to work with -M so I just added another module to t/lib to keep it simple.
I totally understand what you're saying about the implementation details.
It hadn't occurred to me that I didn't need the same dir (that recreating it the same way would result in the same contents).
Thanks for the tip, and the handy module.

**Comment by kentfredric** [comment]
I'd consider adding some sort of hack to make `-M` loading easier, but the syntax it would require eludes me. Even if I say "Hey, stick it in ENV{TFSD_MAP} somehow" it would be potentially very ugly.

**Comment by rwstauner** [comment]
Indeed.
I did get `PERL5OPT` to work, it's just slightly more difficult than an actual `-M` because `PERL5OPT` gets split on spaces.  But jamming everything together does work (if you don't follow the module name with an equal sign):
`PERL5OPT='-MTest::File::ShareDir::Dist{"foo"=>"bar"}'`

At this point I'm happy with what I've got.
Ideally my use case is rare (i should use a `MyApp->run(@ARGV)` module instead of a script and then i wouldn't need this hack (or be testing with `qx`), but this is an old, inherited dist),
so maybe if it comes up again a brief suggestion in the docs would be sufficient then.
